src/main/java/bio/terra/workspace/app/controller/WorkspaceApiController.java

```diff
@@ -59,6 +67,25 @@ private AuthenticatedUserRequest getAuthenticatedInfo() {
     return new ResponseEntity<WorkspaceDescription>(desc, HttpStatus.OK);
   }
 
+  @Override
+  public ResponseEntity<JobModel> createDataReference(
+      @PathVariable("id") String id, @RequestBody CreateDataReferenceRequestBody body) {
+    AuthenticatedUserRequest userReq = getAuthenticatedInfo();
+    createDataReferenceService.createDataReference(id, body, userReq);
+    // Look up the newly-created job
+    JobModel createJob = jobService.retrieveJob(body.getJobControl().getJobid(), userReq);
```

Comment:
it seems like this might not need to be an async job anymore based on some discussion in appservices slack, but I had already implemented it as such so leaving it for now.